topologytest/hlv_test.go

```diff
@@ -93,22 +158,64 @@ func TestHLVUpdateDocumentSingleActor(t *testing.T) {
 			} else {
 				t.Skip("intermittent failure in Couchbase Server CBG-4329")
 			}
+			if base.UnitTestUrlIsWalrus() {
+				t.Skip("rosmar failure CBG-4365")
+			}
 			peers, _ := setupTests(t, tc.topology, tc.activePeerID)
 
 			body1 := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s", "write": 1}`, tc.activePeerID, tc.description()))
-			createVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(), body1)
+			createVersion := peers[tc.activePeerID].CreateDocument(tc.collectionName(), tc.docID(tc.activePeerID), body1)
 
 			waitForVersionAndBody(t, tc, peers, createVersion, body1)
 
 			body2 := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s", "write": 2}`, tc.activePeerID, tc.description()))
-			updateVersion := peers[tc.activePeerID].WriteDocument(tc.collectionName(), tc.docID(), body2)
+			updateVersion := peers[tc.activePeerID].WriteDocument(tc.collectionName(), tc.docID(tc.activePeerID), body2)
 			t.Logf("createVersion: %+v, updateVersion: %+v", createVersion, updateVersion)
 			t.Logf("waiting for document version 2 on all peers")
 			waitForVersionAndBody(t, tc, peers, updateVersion, body2)
 		})
 	}
 }
 
+func TestHLVUpdateDocumentMultiActor(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyCRUD, base.KeyImport, base.KeyVV)
+
+	for _, tc := range getMultiActorTestCases() {
+		t.Run(tc.description(), func(t *testing.T) {
+			if strings.Contains(tc.description(), "CBL") {
+				t.Skip("Skipping Couchbase Lite test, returns unexpected body in proposeChanges: [304], CBG-4257")
+			}
+			if base.UnitTestUrlIsWalrus() {
+				t.Skip("rosmar consistent failure CBG-4365")
+			} else {
+				t.Skip("intermittent failure in Couchbase Server CBG-4329")
+			}
+			peers, _ := setupTests(t, tc.topology, "")
+
+			for peerName, _ := range peers {
+				tc.originalActor = peerName
+				docID := tc.docID(tc.getOriginalActor())
+				body1 := []byte(fmt.Sprintf(`{"originPeer": "%s", "topology": "%s", "writePeer": "%s"}`, peerName, tc.description(), peerName))
+				createVersion := peers[peerName].CreateDocument(tc.collectionName(), docID, body1)
+				waitForVersionAndBody(t, tc, peers, createVersion, body1)
+
+				peerList := maps.Keys(peers)
```

Comment:
On the single actor tests I put `PeerNames` on the interface, and sorted them to provide some determinism when writing the tests. I think putting this on the general interface will shorten where you are iterating over peers in each test.

We could then decide later if we wanted to do a different order.